src/CADL.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -512,15 +511,25 @@ export function getInputType(
                     isReadOnly = true;
                 }
                 if (isNeverType(value.type) || isVoidType(value.type)) return;
+                const jsonKey = "json";
+                const csharpKey = "csharp";
+                const clientKey = "client";
+                const projectedNamesMap = getProjectedNames(program, value);
+                const name =
+                    projectedNamesMap?.get(csharpKey) ??
+                    projectedNamesMap?.get(clientKey) ??
+                    value.name;
+                const serializedName =
+                    projectedNamesMap?.get(jsonKey) ?? value.name;
                 const inputProp = {
-                    Name: value.name,
-                    SerializedName: value.name,
+                    Name: name,
+                    SerializedName: serializedName,
```

Comment:
What if projectName and friendlyname exist together, I think friendlyname wins.

Reply:
Do we have a place to search the "target" of those decorator? Currently we only handles `@friendlyName` on **models**. But `@projectedName` works on **properties**.
I do not know if `@friendlyName` could work on properties, or `@projectedName` could work on models.
If both not, we do not need to worry about it.

Reply:
The definition of `@friendlyName` is 
```
export function $friendlyName(
  context: DecoratorContext,
  target: Type,
  friendlyName: string,
  sourceObject: Type | undefined
)
```,
so it could be applied on any type.

Reply:
@pshao25,  could you share the doc link of @friendlyName?    I am unable to find it in the latest spec.

Reply:
> 

this makes sense but this is not the goal of my PR, therefore I will not add support friendlyName for properties in this PR.
We should do not in a separate PR, and we handle "what should happen if a property has both" then.

Reply:
Double checked we only applied on models, so agreed to not consider this corner case now.

Reply:
> @pshao25, could you share the doc link of @friendlyName? I am unable to find it in the latest spec.

https://github.com/microsoft/typespec/blob/ec21edc66926da0541a4d645be4ccd9f260add01/packages/compiler/lib/decorators.ts#L708-L712

Reply:
Resolving this because we currently did not support friendlyName on properties.